src/crawlee/abstract_http_crawler/_abstract_http_crawler.py

```diff
@@ -92,11 +95,34 @@ def _create_static_content_crawler_pipeline(self) -> ContextPipeline[ParsedHttpC
         """Create static content crawler context pipeline with expected pipeline steps."""
         return (
             ContextPipeline()
+            .compose(self._execute_pre_navigation_hooks)
```

Comment:
It does not need to be it's own pipeline step. It can be part of _make_http_request. This makes it maybe more readable.

Reply:
I like it this way.